doc/source/development/window.rst

```diff
@@ -0,0 +1,242 @@
+Generalized Window Operations
+=============================
+
+Rationale
+---------
+The current window implementation in :file:`window.pyx` is difficult to
+customize and couples the creation of window bounds, loop over the values to
+aggregate, the logic for updating an aggregation. The aggregation algorithms
+are also duplicated in :file:`groupby.pyx`, but unifying grouping aggregation
```

Comment:
We should break the last sentence here into its own paragraph, and precede it with something like 'This is a design specification for a generic implementation of windowed operations. It is meant to replace existing windowing implementations, as well as allow users to customize windowing behavior.'